core/shared/src/main/scala-2/zio/ZLayerCompanionVersionSpecific.scala

```diff
@@ -53,7 +53,7 @@ private[zio] trait ZLayerCompanionVersionSpecific {
    * val carLayer: URLayer[Wheels & Engine, Car] = ZLayer.derive[Car]
    * }}}
    */
-  def derive[A]: ZLayer[Nothing, Any, A] = macro zio.internal.macros.ZLayerDerivationMacros.deriveImpl[A]
+  def derive[A]: ZLayer[Any, Nothing, A] = macro zio.internal.macros.ZLayerDerivationMacros.deriveImpl[A]
```

Comment:
Shouldn't we use `[Any, Throwable, A]` instead? AFAIK the derivation macros also search for implicit Config which might fail

Reply:
Actually, my bad, I don't think having `Any` as the environment can work since we'll almost always have a service as a requirement. I'll move the discussion on the issue instead

Reply:
(Copied from #8889)

This type is chosen on purpose as the upper bound of ZLayer emitting A. ([Scala 3 Guide: Transparent Inline Methods](https://docs.scala-lang.org/scala3/guides/macros/inline.html#transparent-inline-methods))

If it affects type inference, maybe the Nothing part should be replaced with ZNothing, but I'm not 100% sure.